### PR TITLE
NextJS (React) UI for zkApps review.

### LINKED ISSUE #754: HF audit zkApp tutorials: verify and run through code for Tutorial 4: Build a zkApp UI in the Browser with React
HF audit zkApp tutorials: verify and run through code for Tutorial 4: Build a zkApp UI in the Browser with React

New issue to verify that the code runs for this tutorial as part of the tutorial audit
Earlier issues may not have satisfied the Definition of Done (DoD) where the code is run against the latest versions of o1js and zkApp CLI

### PR DESCRIPTION
**Pull request by shimkiv**
Closes #754

### CODE REVIEW

File: examples/zkapps/04-zkapp-browser-ui/ui/src/pages/index.page.tsx [hunk 1]
```diff
@@ -17,7 +17,7 @@ export default function Home() {
     currentNum: null as null | Field,
     publicKey: null as null | PublicKey,
     zkappPublicKey: null as null | PublicKey,
-    creatingTransaction: false
+    creatingTransaction: false,
```
Comment (MartinMinkov) [inline examples/zkapps/04-zkapp-browser-ui/ui/src/pages/index.page.tsx:20]: Could you clarify whether these formatting changes cover every tutorial or are just specific to these files changed in the PR? If it's only these files, could you apply the same formatting to all remaining files? Thanks!

**Review by MartinMinkov — APPROVED** [review]

File: examples/zkapps/04-zkapp-browser-ui/ui/src/pages/index.page.tsx (hunk 1 above)
Comment (ymekuria) [reply examples/zkapps/04-zkapp-browser-ui/ui/src/pages/index.page.tsx:20]: Agreed. Are theses just prettier formatting updates?

Comment (shimkiv) [reply examples/zkapps/04-zkapp-browser-ui/ui/src/pages/index.page.tsx]: Reverted.

File: docs/zkapps/tutorials/02-private-inputs-hash-functions.mdx [hunk 2]
```diff
@@ -39,7 +39,7 @@ In this tutorial, you build a smart contract with a piece of private state that
 
 ## Prerequisites
 
-This tutorial has been tested with [Mina zkApp CLI](https://github.com/o1-labs/zkapp-cli) version `0.16.0`.
+This tutorial has been tested with [zkApp CLI](https://www.npmjs.com/package/zkapp-cli) version `0.16.0`.
```
[the remark this answers is not in the document]
Comment (ymekuria) [reply docs/zkapps/tutorials/02-private-inputs-hash-functions.mdx:42]: Actually 0.20.1.

**Review by ymekuria — APPROVED** [review]
Where you able to successfully run the code in examples/04-zkapp-browser-ui and send transactions with UI and auro wallet?

**Comment by shimkiv** [comment]
> @shimkiv Where you able to successfully run the code in examples/04-zkapp-browser-ui and send transactions with UI and auro wallet?

Of course @ymekuria, check the updated document, configuration section:
https://github.com/o1-labs/docs2/pull/915/files#diff-ffc9c319ebf45bddd9d67b2510c54e40fbab10ffb8a42283f3e6d43cd6394473R176

I checked it against the lightnet not the Berkeley nor Devnet, so it works, there might be some troubles with Berkeley and Devnet at the moment but should be fine after the HF.

---

Tools versions will be updated by another PR in review.

**Comment by ymekuria** [comment]
> > @shimkiv Where you able to successfully run the code in examples/04-zkapp-browser-ui and send transactions with UI and auro wallet?
> 
> Of course @ymekuria, check the updated document, configuration section: https://github.com/o1-labs/docs2/pull/915/files#diff-ffc9c319ebf45bddd9d67b2510c54e40fbab10ffb8a42283f3e6d43cd6394473R176
> 
> I checked it against the lightnet not the Berkeley nor Devnet, so it works, there might be some troubles with Berkeley and Devnet at the moment but should be fine after the HF.
> 
> Tools versions will be updated by another PR in review.

The only reason I asked was because someone on discord was having issues running the tutorial before the audit.

**Pull request merged**